### PR TITLE
Add hover effects for `MacosIconButton` and `MacosBackButton` (closes #168)

### LINKED ISSUE #168: 💡 Add hover effects for MacosIconButtons and MacosBackButtons
## Use case

Currently, when the user hovers over a `MacosIconButton` or similar, there is no visual feedback to show that the element is interactive. Previously, the cursor changed to a hand cursor, but this was changed in 0.9.2 to better match Apple guidelines.

## Proposal

Add a visual cue, like a drop shadow, to the `MacosIconButton` and `MacosBackButton`, when the user's mouse cursor hovers over them. 

This is already used in Apple UI design, for example in Notes:

https://user-images.githubusercontent.com/9117427/137474637-846f20cb-cd5b-4da0-be89-516d7c349f8c.mp4

or in Safari (with a dark theme):

https://user-images.githubusercontent.com/9117427/137474879-8c7e2eaf-2967-49af-80ee-f67123babd48.mov

### PR DESCRIPTION
**Pull request by whiplashoo**
This adds subtle hover effects (background color change) for both widgets, to better match macOS design norms. 

Adds a `hoverColor` property to `MacosIconButton` and `MacosBackButton`. The default hover color values were taken by apps like Notes, Reminders, and Safari, which feature similar icon buttons. In the example screen, you can see the hover effect:

https://user-images.githubusercontent.com/9117427/142167916-28be9764-4914-4eea-b2a9-88c4affb6e65.mov

Finally, while it was not necessary, the default shape of `MacosIconButton` has been changed from `BoxShape.circle` to `BoxShape.rectangle` with `BorderRadius(circular: 7)`. After looking around at other apps in macOS, almost every icon button has a rounded rectangular shape.

## Pre-launch Checklist

- [✅] I have run `dartfmt` on all changed files 
- [✅] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [✅] I have added/updated relevant documentation 
- [✅] I have run "optimize/organize imports" on all changed files
- [✅] I have addressed all analyzer warnings as best I could

### CODE REVIEW

File: example/macos/Podfile.lock [hunk 1]
```diff
@@ -19,4 +19,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 6eac6b3292e5142cfc23bdeb71848a40ec51c14c
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.10.2
```
Comment (GroovinChip) [inline example/macos/Podfile.lock]:
```suggestion
COCOAPODS: 1.11.2
```

File: lib/src/buttons/back_button.dart [hunk 2]
```diff
@@ -21,6 +22,9 @@ class MacosBackButton extends StatefulWidget {
   /// The color to fill the space around the icon with.
   final Color? fillColor;
 
+  /// The color of the button's background when the button is hovered (mouse is over the button).
```
Comment (GroovinChip) [inline lib/src/buttons/back_button.dart]:
```suggestion
  /// The color of the button's background when the mouse hovers over the button.
```

File: lib/src/buttons/back_button.dart [hunk 3]
```diff
@@ -139,8 +146,27 @@ class MacosBackButtonState extends State<MacosBackButton>
...
+          ? const Color(0xff333336)
+          : const Color(0xffF3F2F2);
+    }
+
     return MouseRegion(
       cursor: widget.mouseCursor!,
+      onEnter: (e) {
+        setState(() {
+          _isHovered = true;
+        });
```
Comment (GroovinChip) [inline lib/src/buttons/back_button.dart]:
```suggestion
        setState(() => _isHovered = true);
```

File: lib/src/buttons/back_button.dart [hunk 4]
```diff
@@ -139,8 +146,27 @@ class MacosBackButtonState extends State<MacosBackButton>
...
+      },
+      onExit: (e) {
+        setState(() {
+          _isHovered = false;
+        });
```
Comment (GroovinChip) [inline lib/src/buttons/back_button.dart]:
```suggestion
        setState(() => _isHovered = false);
```

File: lib/src/buttons/icon_button.dart [hunk 5]
```diff
@@ -41,6 +42,9 @@ class MacosIconButton extends StatefulWidget {
   /// The color of the button's background when the button is disabled.
   final Color? disabledColor;
 
+  /// The color of the button's background when the button is hovered (mouse is over the button). Set to Colors.transparent to disable the hover effect.
```
Comment (GroovinChip) [inline lib/src/buttons/icon_button.dart]:
```suggestion
  /// The color of the button's background when the mouse hovers over it. 
  /// 
  /// Set to `Colors.transparent` to disable the hover effect.
```

File: lib/src/buttons/icon_button.dart [hunk 6]
```diff
@@ -55,12 +59,13 @@ class MacosIconButton extends StatefulWidget {
...
   ///
-  /// Defaults to `BoxShape.circle`.
+  /// Defaults to `BoxShape.rectangle`.
   final BoxShape shape;
 
   /// The border radius for the button.
   ///
-  /// This should only be set if setting [shape] to `BoxShape.rectangle`.
+  /// This should only be set if setting [shape] to `BoxShape.rectangle` and defaults
+  /// to `BorderRadius.circular(7.0)`.
```
Comment (GroovinChip) [inline lib/src/buttons/icon_button.dart]:
```suggestion
  /// This should only be set if setting [shape] to `BoxShape.rectangle`. 
  ////
  /// Defaults to `BorderRadius.circular(7.0)`.
```

File: lib/src/buttons/icon_button.dart [hunk 7]
```diff
@@ -206,6 +216,16 @@ class MacosIconButtonState extends State<MacosIconButton>
 
     return MouseRegion(
       cursor: widget.mouseCursor!,
+      onEnter: (e) {
+        setState(() {
+          _isHovered = true;
+        });
```
Comment (GroovinChip) [inline lib/src/buttons/icon_button.dart]:
```suggestion
        setState(() => _isHovered = true);
```

File: lib/src/buttons/icon_button.dart [hunk 8]
```diff
@@ -206,6 +216,16 @@ class MacosIconButtonState extends State<MacosIconButton>
...
+      },
+      onExit: (e) {
+        setState(() {
+          _isHovered = false;
+        });
```
Comment (GroovinChip) [inline lib/src/buttons/icon_button.dart]:
```suggestion
        setState(() => _isHovered = false);
```

File: lib/src/buttons/icon_button.dart [hunk 9]
```diff
@@ -312,6 +340,8 @@ class MacosIconButtonThemeData with Diagnosticable {
   /// The default background color for [MacosIconButton].
   final Color? backgroundColor;
 
+  final Color? hoverColor;
```
Comment (GroovinChip) [inline lib/src/buttons/icon_button.dart:343]:
```suggestion
  /// The color of the button when the mouse hovers over it.
  final Color? hoverColor;
```

**Review by GroovinChip — CHANGES_REQUESTED** [review]

**Comment by whiplashoo** [comment]
Updated per the requested changes. I also fixed a failing test for `MacosIconButton`. 

Still failing the GH-pages build with a 403, even though I accepted your invitation before.

**Comment by GroovinChip** [comment]
> Still failing the GH-pages build with a 403, even though I accepted your invitation before.

Yes, this is because you need to clone my repo directly, and not use your fork. You can start doing that once we get both your PR's merged. Review coming soon.

**Review by GroovinChip — APPROVED** [review]
LGTM

**Pull request merged**